### PR TITLE
Enforce upper limit of SETTINGS_INITIAL_WINDOW_SIZE 

### PR DESCRIPTION
**Pull request by johanandren**
Refs #773

### CODE REVIEW

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala [hunk 1]
```diff
@@ -560,6 +560,13 @@ class Http2ServerSpec extends AkkaSpec("" + "akka.loglevel = debug")
 
         expectSettingsAck() // TODO check that the setting was indeed applied
       }
+
+      "react on too large SETTINGS_INITIAL_WINDOW_SIZE with FLOW_CONTROL_ERROR" in new TestSetup with RequestResponseProbes {
+        sendSETTING(SettingIdentifier.SETTINGS_INITIAL_WINDOW_SIZE, Math.pow(2, 32).toInt) // valid values are below 2^32 - 1)
```
Comment (johanandren) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala]: Dang, it was supposed to be 2^31

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala [hunk 2]
```diff
@@ -178,10 +178,18 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
...
             case SettingsFrame(settings) ⇒
               if (settings.nonEmpty) debug(s"Got ${settings.length} settings!")
+
+              var settingsAppliedOk = true
+
               settings.foreach {
                 case Setting(Http2Protocol.SettingIdentifier.SETTINGS_INITIAL_WINDOW_SIZE, value) ⇒
-                  debug(s"Setting initial window to $value")
-                  multiplexer.updateDefaultWindow(value)
+                  if (value <= Http2Protocol.MaxInitialWindowSize) {
```
Comment (jrudolph) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: Isn't the value `Int.MaxValue`? So, the condition will be trivially true. I think the sign needs to be checked in the parser instead.

Comment (johanandren) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: `Int.MaxValue - 1` actually

Comment (johanandren) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: I take that back, it is exactly Int.MaxValue, you are right

Comment (johanandren) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: Will we be able to reply with a FLOW_CONTROL_ERROR if the check is in the parser?

Comment (jrudolph) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: Yes, by throwing the right exception, there should be similar things checked there.

Comment (johanandren) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: When I think of it, can't I just as well check the sign here?

Comment (jrudolph) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: Yes, I guess so. I thought we had precedent of checking other settings in the parser but it seems we don't. Go for it ;)

**Comment by jrudolph** [comment]
Thanks, @johanandren, LGTM.

**Pull request merged**